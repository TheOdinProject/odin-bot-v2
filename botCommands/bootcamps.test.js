const command = require('./bootcamps');

describe('?bootcamps', () => {
  describe('regex', () => {
    it.each([
      ['?bootcamps'],
      [' ?bootcamps'],
      ['?bootcamps @odin-bot'],
      ['@odin-bot ?bootcamps'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['botcamps'],
      ['bootcamps'],
      ['bootcamp'],
      ['?bootcamp'],
      ['?bootcmp'],
      ['? bootcamps'],
      ['?abootcamps'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! ?bootcamps'],
      ['Don\'t worry about ?bootcamps'],
      ['Hey @odin-bot, ?bootcamps'],
      ['?@odin-bot ^ ?me ?bootcamps ?tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user?bootcamps'],
      ['it\'s about?bootcamps'],
      ['?bootcampsanillusion'],
      ['?bootcamps?'],
      ['?bootcamps*'],
      ['?bootcamps...'],
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot();
    });
  });
});
