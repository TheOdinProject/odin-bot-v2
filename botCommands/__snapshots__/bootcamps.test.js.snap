// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`?bootcamps callback returns correct output 1`] = `"Friends don't let friends commit to bootcamps without being informed.  https://twitter.com/lzsthw/status/1212284566431576069"`;
